### ISSUE TITLE
Disable device discovery scanning when on a large network

### ISSUE THREAD

**Issue by unwiredben** [rokucommunity/vscode-brightscript-language#500]
The current default for device discovery is fine when a developer is on a small network, but it can cause issues with network traffic on a larger network.  I suggest that the default be based on the current network's netmask -- something with a /24 netmask (like a home network on 192.168.1.xxx) would be OK to default to scanning, while a /16 or /8 network would be out.

This would be especially useful for developers adopting this plugin at Roku, since one VSCode instance can end up discovering hundreds of devices.

**Comment by TwitchBronBron** [comment]
I believe we already have a setting for this (assuming it isn't buggy...haven't tested it in a while). 

brightscript.deviceDiscovery.enabled

**Comment by unwiredben** [comment]
Yes, the setting already exists, but it defaults to enabled.  As developers at Roku independently find this tool, we've got to constantly remind them to change this setting.  That's why I wanted it to be smart about when it should be default-enabled.
